e2e/util/namespace.go

```diff
@@ -49,20 +75,56 @@ func DeleteNamespace(ctx types.Context, cluster types.Cluster, name string) erro
 		},
 	}
 
-	err := cluster.Client.Delete(ctx.Context(), ns)
+	managed, err := isManagedByRamenE2e(ctx, cluster, ns)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.Debugf("Namespace %q not found in cluster %q", name, cluster.Name)
+
+			return nil
+		}
+
+		return err
+	}
+
+	if !managed {
+		log.Debugf("Namespace %q in cluster %q is not managed by ramen-e2e, skipping deletion.",
+			name, cluster.Name)
```

Comment:
Lets change to:

    "Skipping deletion of namespace %q: not managed by ramen-e2e (missing label %s=%s)"